CBLClient/Apps/CBLTestServer-Android/app/src/main/java/com/couchbase/CouchbaseLiteServ/server/RequestHandler/LoggingRequestHandler.java

```diff
@@ -0,0 +1,88 @@
+package com.couchbase.CouchbaseLiteServ.server.RequestHandler;
+
+import android.content.Context;
+
+import com.couchbase.CouchbaseLiteServ.MainActivity;
+import com.couchbase.CouchbaseLiteServ.server.Args;
+import com.couchbase.lite.Database;
+import com.couchbase.lite.LogFileConfiguration;
+import com.couchbase.lite.LogLevel;
+
+import java.io.File;
+
+public class LoggingRequestHandler {
+    /* ----------- */
+    /* - Logging - */
+    /* ----------- */
+
+    public String configure(Args args){
+        String log_level = args.get("log_level");
+        String directory = args.get("directory");
+        int maxRotateCount = args.get("max_rotate_count");
+        long maxSize = args.get("max_size");
+        boolean plainText = args.get("plain_text");
+
+        if (directory.isEmpty()) {
+            Context context = MainActivity.getAppContext();
+            long ts = System.currentTimeMillis()/1000;
+            directory = context.getFilesDir().getAbsolutePath() + "/logs_" + ts ;
+
+            System.out.println("File logging configured at: " + directory.toString());
+        }
+        LogFileConfiguration config = new LogFileConfiguration(directory);
+        if (maxRotateCount > 1) {
+            config.setMaxRotateCount(maxRotateCount);
+        }
+        if (maxSize > 512000) {
+            config.setMaxSize(maxSize);
+        }
+        config.setUsePlaintext(plainText);
+        Database.log.getFile().setConfig(config);
+        switch (log_level) {
+            case "debug":
+                Database.log.getFile().setLevel(LogLevel.DEBUG);
+                break;
+            case "verbose":
+                Database.log.getFile().setLevel(LogLevel.VERBOSE);
+                break;
+            case "info":
+                Database.log.getFile().setLevel(LogLevel.INFO);
+                break;
+            case "error":
+                Database.log.getFile().setLevel(LogLevel.ERROR);
+                break;
+            case "warning":
+                Database.log.getFile().setLevel(LogLevel.WARNING);
+                break;
+            default:
+                Database.log.getFile().setLevel(LogLevel.NONE);
+                break;
+        }
+        return directory;
+    }
+
+    public boolean getPlainTextStatus(Args args) {
```

Comment:
can you also add set methods for all get methods and add an API to create a config

Reply:
configure method creates config and also use all the set. I can create separate set methods if it's needed.

Reply:
having set methods will help in changing/modifying the config once config is created. it may not be required now, it might require later. Also it is one of the best coding practices  if we have set methods along with get methods

Reply:
added.